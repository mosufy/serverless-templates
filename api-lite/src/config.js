export const app = {
  env: process.env.APP_ENV || process.env.NODE_ENV,
  debug: process.env.APP_DEBUG === 'true',
};

export default {
  app,
};
