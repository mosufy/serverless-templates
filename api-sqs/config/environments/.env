APP_ENV="local"
APP_DEBUG=true

AWS_ACCOUNT_REGION="us-west-1"
AWS_ACCOUNT_PROFILE="slsDevProfile"
AWS_ACCOUNT_ID=

AWS_LAMBDA_TIMEOUT=3
AWS_LAMBDA_MEMORY_SIZE=128

AWS_LOG_RETENTION_DAYS=7

AWS_APIGATEWAY_SECRET_KEY=
AWS_APIGATEWAY_COMPRESSION_MAX_BYTES=100
