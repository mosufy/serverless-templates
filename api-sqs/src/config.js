export const app = {
  env: process.env.APP_ENV || process.env.NODE_ENV,
  debug: process.env.APP_DEBUG === 'true',
};

const awsSQSPingQueueName = `sls-tpl-api-sqs-${app.env}-pingQueue`;

export const aws = {
  sqs: {
    options: {
      override: process.env.AWS_SQS_OPTIONS_OVERRIDE === 'true',
      accessKeyId: process.env.AWS_SQS_OPTIONS_ACCESS_KEY_ID,
      secretAccessKey: process.env.AWS_SQS_OPTIONS_SECRET_ACCESS_KEY,
      region: process.env.AWS_SQS_OPTIONS_REGION,
    },
    pingQueue: {
      name: awsSQSPingQueueName,
      url: `https://sqs.${process.env.AWS_ACCOUNT_REGION}.amazonaws.com/${process.env.AWS_ACCOUNT_ID}/${awsSQSPingQueueName}`,
    },
  },
};

export default {
  app,
  aws,
};
