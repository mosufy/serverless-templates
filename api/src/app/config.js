const config = {
  app: {
    env: process.env.APP_ENV,
    debug: process.env.APP_DEBUG,
  },
};

export default config;
