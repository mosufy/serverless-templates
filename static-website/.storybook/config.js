import { configure, addParameters, addDecorator } from "@storybook/react";
import { withInfo } from "@storybook/addon-info";

addParameters({
  options: {
    isFullscreen: false,
    showAddonsPanel: true,
    showSearchBox: false,
    panelPosition: "bottom",
    sidebarAnimations: false,
    enableShortcuts: true,
  },
});

addDecorator(
  withInfo({
    inline: true,
  })
);

configure(require.context("../stories/", true, /\.stories\.(js|mdx)$/), module);
