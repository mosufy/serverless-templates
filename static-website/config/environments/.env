APP_ENV="local"
APP_DEBUG=true

AWS_ACCOUNT_REGION="us-west-1"
AWS_ACCOUNT_PROFILE="slsDevProfile"

CLIENT_S3_BUCKETNAME="serverless-templates-static-website"
CLIENT_APP_DISTFOLDER="build"
CLIENT_CF_ID=

# React environments
REACT_APP_ENV=${APP_ENV}
REACT_APP_DEBUG=${APP_DEBUG}