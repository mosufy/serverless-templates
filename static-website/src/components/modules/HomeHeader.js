import React from "react";

import Logo from "Components/atoms/Logo";

const HomeHeader = () => {
  return (
    <header className="App-header">
      <Logo />
      <p>
        Edit <code>src/components/App.js</code> and save to reload.
      </p>
      <a
        className="App-link"
        href="https://reactjs.org"
        target="_blank"
        rel="noopener noreferrer">
        Learn React
      </a>
    </header>
  );
};

export default HomeHeader;
