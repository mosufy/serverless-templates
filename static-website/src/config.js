export const app = {
  env: process.env.REACT_APP_ENV || process.env.NODE_ENV,
  debug: process.env.REACT_APP_DEBUG === "true",
};

export default {
  app,
};
