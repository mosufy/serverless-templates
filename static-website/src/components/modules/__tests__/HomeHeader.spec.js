import React from "react";
import { shallow } from "enzyme";

import HomeHeader from "../HomeHeader";
import { app } from "../../../config";

describe("test HomeHeader component", () => {
  it("should render correctly", () => {
    const component = shallow(<HomeHeader />);
    expect(component).toMatchSnapshot();
  });

  it("should render with debug turned off", () => {
    app.debug = false;
    const component = shallow(<HomeHeader />);
    expect(component).toMatchSnapshot();
    app.debug = true;
  });
});
