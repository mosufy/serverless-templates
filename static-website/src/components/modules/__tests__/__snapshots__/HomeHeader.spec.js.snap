// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test HomeHeader component should render correctly 1`] = `
<header
  className="App-header"
>
  <Logo />
  <p>
    Edit 
    <code>
      src/components/App.js
    </code>
     and save to reload.
  </p>
  <a
    className="App-link"
    href="https://reactjs.org"
    rel="noopener noreferrer"
    target="_blank"
  >
    Learn React
  </a>
  <ul
    style={
      Object {
        "fontSize": "16px",
        "listStyleType": "none",
      }
    }
  >
    <li>
      APP_ENV: 
      test
    </li>
    <li>
      APP_DEBUG: 
      true
    </li>
  </ul>
</header>
`;

exports[`test HomeHeader component should render with debug turned off 1`] = `
<header
  className="App-header"
>
  <Logo />
  <p>
    Edit 
    <code>
      src/components/App.js
    </code>
     and save to reload.
  </p>
  <a
    className="App-link"
    href="https://reactjs.org"
    rel="noopener noreferrer"
    target="_blank"
  >
    Learn React
  </a>
  <ul
    style={
      Object {
        "fontSize": "16px",
        "listStyleType": "none",
      }
    }
  >
    <li>
      APP_ENV: 
      test
    </li>
    <li>
      APP_DEBUG: 
      false
    </li>
  </ul>
</header>
`;
